src/schema/v2/conversation/index.ts

```diff
@@ -479,6 +508,61 @@ export const ConversationType = new GraphQLObjectType<any, ResolverContext>({
     },
 
     messagesConnection,
+
+    messagesAndConversationEventsConnection: {
+      type: MessagesAndConversationEventsConnection,
+      description:
+        "A connection for all messages and events in a single conversation",
+      args: pageable(),
+      resolve: (
+        { id, from_email, initial_message, from_name, embedded },
+        options,
+        { conversationWithEventsLoader }
+      ) => {
+        if (!conversationWithEventsLoader) return null
+        const optionKeys = Object.keys(options)
+        if (optionKeys.includes("last") && !optionKeys.includes("before")) {
+          options.before = `${lastMessageId({ embedded })}`
+        }
+        const { page, size, offset } = convertConnectionArgsToGravityArgs(
+          options
+        )
+        return conversationWithEventsLoader(id, { page, size }).then(
+          ({ total_count, messages_and_conversation_events }) => {
+            // Inject the convesation initiator's email into each message payload
+            // so we can tell if the user sent a particular message.
+            // Also inject the conversation id, since we need it in some message
+            // resolvers (invoices).
+            /* eslint-disable no-param-reassign */
+            messages_and_conversation_events = messages_and_conversation_events.map(
+              (messageOrEvent) => {
+                if (messageOrEvent.type === "message_detail") {
+                  return merge(messageOrEvent, {
+                    conversation_initial_message: initial_message,
+                    conversation_from_name: from_name,
+                    conversation_from_address: from_email,
+                    conversation_id: id,
+                  })
+                } else {
+                  return messageOrEvent
+                }
+              }
+            )
+            return {
+              ...connectionFromArraySlice(
```

Comment:
Is there any chance we can use our [`paginationResolver`](https://github.com/artsy/metaphysics/blob/main/docs/pagination.md) helper here?

Reply:
ooo we totally can! did not know that was a thing. definitely cleans up the code a lot. done in https://github.com/artsy/metaphysics/pull/5168/commits/c22c7b724280a135c90ae2ac2ba639d2c7539a50